### ISSUE TITLE
CPU Hog migrating from 2.8.0 to 3.1.0

### ISSUE THREAD

**Issue by rodri042** [Unitech/pm2#3914]
## What's going wrong?
I'm using [node-gameloop](https://www.npmjs.com/package/node-gameloop) to create a precise loop for my game server. When I run my code with `node` and `pm2@2.8.0` everything goes fine, but with `pm2@3.1.0` cpu goes 6% and 7% all time and the loop is not precise.

Video showing the issue: https://www.youtube.com/watch?v=kSIwH1AYmBY

## How could we reproduce this issue?
Test code:
- `npm install node-gameloop`
- 

```js
const gameLoop = require("node-gameloop");

const loop = (delta) => {
  const elapsedMs = Math.floor(delta * 1000);

  if (elapsedMs > 30) {
    console.error("DELTA WAS", elapsedMs, "INSTEAD OF", 30);
  }
};

const physicsLoop = gameLoop.setGameLoop(loop, 30);
const updateLoop = gameLoop.setGameLoop(() => {}, 40);
```

- Run with pm2 2.8.0 and you should'nt see the error logs (or very few of them) and `pm2 list` shows 0% all time.
- Run with pm2 3.1.0 and you should often see the error logs and `pm2 list` shows 6% cpu all time.

## Supporting information

```
===============================================================================
--- PM2 REPORT (Fri Sep 14 2018 15:46:13 GMT-0300 (-03)) ----------------------
===============================================================================
--- Daemon -------------------------------------------------
pm2d version         : 3.1.2
node version         : 6.11.5
node path            : /usr/local/bin/pm2
argv                 : /home/rodri042/.nvm/versions/node/v6.11.5/bin/node,/usr/local/lib/node_modules/pm2/lib/Daemon.js
argv0                : node
user                 : rodri042
uid                  : 1000
gid                  : 1000
uptime               : 58min
===============================================================================
--- CLI ----------------------------------------------------
local pm2            : 3.1.2
node version         : 6.11.5
node path            : /usr/local/bin/pm2
argv                 : /home/rodri042/.nvm/versions/node/v6.11.5/bin/node,/usr/local/bin/pm2,report
argv0                : node
user                 : rodri042
uid                  : 1000
gid                  : 1000
===============================================================================
--- System info --------------------------------------------
arch                 : x64
platform             : linux
type                 : Linux
cpus                 : Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
cpus nb              : 4
freemem              : 470589440
totalmem             : 8264663040
home                 : /home/rodri042
===============================================================================
--- PM2 list -----------------------------------------------
┌──────────┬────┬──────┬───────┬─────────┬─────────┬────────┬──────┬───────────┬──────────┬──────────┐
│ App name │ id │ mode │ pid   │ status  │ restart │ uptime │ cpu  │ mem       │ user     │ watching │
├──────────┼────┼──────┼───────┼─────────┼─────────┼────────┼──────┼───────────┼──────────┼──────────┤
│ index    │ 0  │ fork │ 0     │ stopped │ 0       │ 0      │ 0%   │ 0 B       │ rodri042 │ disabled │
│ test     │ 1  │ fork │ 11414 │ online  │ 1       │ 19s    │ 6.2% │ 71.7 MB   │ rodri042 │ disabled │
└──────────┴────┴──────┴───────┴─────────┴─────────┴────────┴──────┴───────────┴──────────┴──────────┘
===============================================================================
--- Daemon logs --------------------------------------------
┌──────────┬────┬──────┬───────┬─────────┬─────────┬────────┬─────┬───────────┬──────────┬──────────┐
│ App name │ id │ mode │ pid   │ status  │ restart │ uptime │ cpu │ mem       │ user     │ watching │
├──────────┼────┼──────┼───────┼─────────┼─────────┼────────┼─────┼───────────┼──────────┼──────────┤
│ index    │ 0  │ fork │ 0     │ stopped │ 0       │ 0      │ 0%  │ 0 B       │ rodri042 │ disabled │
│ test     │ 1  │ fork │ 11414 │ online  │ 1       │ 19s    │ 0%  │ 71.7 MB   │ rodri042 │ disabled │
└──────────┴────┴──────┴───────┴─────────┴─────────┴────────┴─────┴───────────┴──────────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```

**Comment by julbra** [comment]
I confirm the same on all my systems where the average CPU load jumped up by 6%.

**Comment by Unitech** [comment]
Fixed via PM2 (3.2.0):

```bash
$ npm install pm2 -g
$ pm2 update
```

Blog post: https://blog.pm2.io/pm2-runtime-3-2-autumn-features/
Changelog: https://github.com/Unitech/pm2/releases/tag/3.2.0
